packages/app-staking/src/Account/BondExtra.tsx

```diff
@@ -83,16 +109,79 @@ class BondExtra extends React.PureComponent<Props, State> {
             className='medium'
             help={t('The maximum amount to increase the bonded value, this is adjusted using the available free funds on the account.')}
             label={t('max additional value')}
+            maxValue={maxBalance}
             onChange={this.onChangeValue}
+            withMax
           />
         </Modal.Content>
       </>
     );
   }
 
+  private nextState (newState: Partial<State>): void {
+    this.setState((prevState: State): State => {
+      const { api } = this.props;
+      const { maxAdditional = prevState.maxAdditional, maxBalance = prevState.maxBalance } = newState;
+      const extrinsic = (maxAdditional && maxAdditional.gte(ZERO))
+        ? api.tx.staking.bondExtra(maxAdditional)
+        : null;
+
+      return {
+        maxAdditional,
+        extrinsic,
+        maxBalance
+      };
+    });
+  }
+
+  private setMaxBalance = () => {
+    const { api, system_accountNonce = ZERO, balances_fees = ZERO_FEES, balances_votingBalance = ZERO_BALANCE } = this.props;
```

Comment:
This one is problematic - votingBalance includes the already stashed balance. so basically is is votingBalance - bondedBalance that can be used.

Reply:
... while problematic, the node will adjust for the max funds, however basically what we display won't be that value.